### PR TITLE
Add checkboxes to determine whether or not to send parameter

### PR DESCRIPTION
**Pull request by synbot**
### Acceptance Criteria
1. Each parameter should have a checkbox to determine if it should be sent
2. Each parameter should have an initially blank default value in the form
3. Request logic should be updated to allow sending of blank values

### Additional Notes

Booleans and enums will need a "blank" option.

### CODE REVIEW

**Comment by zpchavez** [comment]
QA should test this pretty thoroughly. There's a lot of potential for things to have broken.

File: application/ui/components/params-list.jsx [hunk 1]
```diff
@@ -68,16 +81,50 @@ module.exports = React.createClass({
         return this.renderParam(param, []);
     },
 
+    includeChanged : function(event)
+    {
+        var values, path, requestActions;
```
Comment (agerbens) [inline application/ui/components/params-list.jsx]: Is values needed here?

File: application/ui/components/params-list.jsx [hunk 2]
```diff
@@ -68,16 +81,50 @@ module.exports = React.createClass({
...
+                        type      = "checkbox"
+                        name      = {'include-' + param.name}
+                        data-path = {path[0]}
+                        onChange  = {this.includeChanged}
+                        checked   = {! _(this.state.excludedFields).contains(path[0])}
+                    />
+                </td>
+            );
+        } else {
+            checkboxComponent = null;
```
Comment (agerbens) [inline application/ui/components/params-list.jsx]: Should this just be an empty <td> since the header doesn't have this conditional logic?

**Comment by agerbens** [comment]
:interrobang: Left a couple of comments

File: application/ui/components/params-list.jsx (hunk 2 above)
Comment (zpchavez) [inline application/ui/components/params-list.jsx]: I don't think so. If path.length is greater than 1 it means it's a sub-param of a hash or array field.

**Comment by zpchavez** [comment]
:star:

**Comment by baohx2000** [comment]
Should optional params be not included (not checked) by default?

**Comment by zpchavez** [comment]
I don't know. It sort of makes sense but doesn't seem absolutely necessary.

**Comment by baohx2000** [comment]
:pray:

**Comment by pcavagna** [comment]
:+1:

**Comment by zpchavez** [comment]
:three:

**Pull request merged**